CMakeLists.txt

```diff
@@ -60,6 +63,22 @@ if(BUILD_TESTS)
     endif()
 endif()
 
-# Install commands
-install(TARGETS ale LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(DIRECTORY ${ALE_INCLUDE_DIRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+# Generate the package config
+configure_file(cmake/config.cmake.in
+               ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake
+               @ONLY)
+
+# Install the package config
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake
+              DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+# Install the library
+install(TARGETS ale
+        EXPORT aleTargets
+        LIBRARY       DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+# Install the target
+install(EXPORT aleTargets
+        NAMESPACE ale::
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
```

Comment:
This actually writes out the aleTargets target, which creates a file that has the ale target in it.... it's a lot of targets.